### PR TITLE
Add initial draft of magnitude sections

### PR DESCRIPTION
**Pull request by chiphogg**
This involves redirecting the wg21 submodule to my new fork, which simply adds mathjax support.  Users will need to run the appropriate commands to update the submodule.  I don't know what they are, but here is a good guess:

```sh
git submodule sync
git submodule update
```

You'll know things are good when the last commit there is a4dbf650908438148730c238f3d149339569f966.

### CODE REVIEW

File: src/2982R0_std_quantity_as_a_numeric_type.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ audience:
 author:
   - name: Mateusz Pusz ([Epam Systems](http://www.epam.com))
     email: <mateusz.pusz@gmail.com>
+  - name: Chip Hogg ([Aurora Innovation](https://aurora.tech/))
```
Comment (JohelEGP) [inline src/2982R0_std_quantity_as_a_numeric_type.md:10]:
```suggestion
  - name: Charles Hogg ([Aurora Innovation](https://aurora.tech/))
```
Like P2980.

**Review by mpusz — APPROVED** [review]
Thanks!

File: src/2982R0_std_quantity_as_a_numeric_type.md [hunk 2]
```diff
@@ -991,18 +993,136 @@ The above program will produce the following types for acceleration quantities
...
+
+The process also works in reverse: the ratio between any two units of the same dimension is a well-defined
+number.  For example, the ratio between one foot and one inch is 12.
+
 ### Unit magnitudes
 
-TBD
+In principle, this scaling factor can be any positive real number.  In mp-units and Au, we have used the term
+"magnitude" to refer to this scaling factor.  (This should not be confused with the logarithmic "magnitude"
+unit commonly used in astronomy.)
```
Comment (JohelEGP) [inline src/2982R0_std_quantity_as_a_numeric_type.md:1006]: What about these magnitudes?
- https://en.wikipedia.org/wiki/Magnitude_(mathematics)
- https://www.electropedia.org/iev/iev.nsf/display?openform&ievref=102-03-23

I'm exploring just using "vector" for everything that's a vector,
and relying on the abstraction levels to make it clear from context.
Even if I have something like `pos.x`,
where `pos` is of type `cartesian::vector2d`,
and `pos.x` is of type `qty::vector` (a.k.a. `mp_units::quantity` with a scalar number).

File: src/2982R0_std_quantity_as_a_numeric_type.md [hunk 3]
```diff
@@ -991,18 +993,136 @@ The above program will produce the following types for acceleration quantities
...
 
-### Construction helpers
+In the library implementation, each unit is associated with a magnitude.  However, for most units the
```
Comment (JohelEGP) [inline src/2982R0_std_quantity_as_a_numeric_type.md:1008]:
```suggestion
In the library implementation, each unit is associated with a magnitude.  However, for most units, the
```

File: src/2982R0_std_quantity_as_a_numeric_type.md [hunk 4]
```diff
@@ -991,18 +993,136 @@ The above program will produce the following types for acceleration quantities
...
+magnitude is a fully encapsulated implementation detail, not a user-facing value.
 
+This is because the notion of "the" magnitude of a unit is not generally meaningful: it has no physically
+observable consequence.  What _is_ meaningful is the _ratio_ of magnitudes between two units of the same
+dimension.  We could associate the _foot_, say, with any magnitude $m_f$ that we like --- but once we make that
+choice, we must assign $3m_f$ to the _yard_, and $m_f/12$ to the _inch_.  Separately and independently, we can
+assign any magnitude $m_s$ to the second, because it's an independent dimension --- but once we make that
+choice, it fixes the magnitude for derived units, and we must assign, say, $(5280 m_f) / (3600 m_s)$ to the
+_mile per hour_.
```
Comment (JohelEGP) [inline src/2982R0_std_quantity_as_a_numeric_type.md:1017]:
```suggestion
_miles per hour_.
```

File: src/2982R0_std_quantity_as_a_numeric_type.md [hunk 5]
```diff
@@ -991,18 +993,136 @@ The above program will produce the following types for acceleration quantities
...
 
+The one exception to the arbitrariness of magnitudes is _dimensionless_ units.  Because their dimension is
+null, quantities of these units can be meaningfully compared to their squares and other powers.  For example,
```
Comment (JohelEGP) [inline src/2982R0_std_quantity_as_a_numeric_type.md:1020]:
```suggestion
one, quantities of these units can be meaningfully compared to their squares and other powers.  For example,
```

**Review by JohelEGP — COMMENTED** [review]
Guess I'll review it with the rest of the paper.

File: src/2982R0_std_quantity_as_a_numeric_type.md (hunk 4 above)
Comment (chiphogg) [reply src/2982R0_std_quantity_as_a_numeric_type.md:1017]: I think the original is correct here.  "the mile per hour" is grammatically consistent with "the foot" and "the inch" above.

File: src/2982R0_std_quantity_as_a_numeric_type.md [hunk 6]
```diff
@@ -991,18 +993,136 @@ The above program will produce the following types for acceleration quantities
...
+units have two components that participate in operations: _dimension_, and _magnitude_.  When we apply an
+operation to units, we simply apply it separately to the dimension and magnitude parts.
+
+Units are closed under _products_ and _rational powers_.  Therefore, our magnitude representation must support
+these operations natively and robustly; this is the most basic requirement.  We must also support certain
+_irrational_ "ratios", such as the factor of $\frac{\pi}{180}$ between _degrees_ and _radians_.
+
+The usual approach, `std::ratio`, fails to satisfy these requirements in multiple ways.
+
+- It is not closed under rational powers (rather infamously, in the case of $2^\frac{1}{2}$).
```
Comment (JohelEGP) [inline src/2982R0_std_quantity_as_a_numeric_type.md:1037]: I'm changing this to `2<sup>1/2</sup>` (2<sup>1/2</sup>).
This is how it currently looks on my system:

File: src/2982R0_std_quantity_as_a_numeric_type.md [hunk 7]
```diff
@@ -991,18 +993,136 @@ The above program will produce the following types for acceleration quantities
...
+is the largest unit that evenly divides both.  For example, the common unit of the _foot_ and _inch_ is the
+inch.
+
+This also includes instances where neither unit is an integer multiple of the other.  For example, the common
+unit of the meter and the yard does not have a name, but is equivalent to 800 micrometers.  If we call this
+unit `U`, then there are 1250 `U` per meter, and 1143 `U` per yard.
+
+The practical benefit of this definition is that this unit conversion will simply multiply each participating
+quantity by an exact integer.  If we use integer types to begin with, we can continue to use them without
+losing precision.
```
Comment (JohelEGP) [inline src/2982R0_std_quantity_as_a_numeric_type.md:1100]: The second paragraph is talking about another concept.
The flow would suggest it should be the same concept.

**Pull request merged**